docbrown/db/src/loaders.rs

```diff
@@ -152,25 +168,322 @@ pub mod csv {
         let text = "bitcoin/received_000000000001.csv.gz";
         assert!(!r.is_match(&text));
     }
 
     #[test]
     fn regex_match_2() {
         let r = Regex::new(r".+(sent|received)").unwrap();
+        // todo: move file path to data module
         let text = "bitcoin/sent_000000000001.csv.gz";
         assert!(r.is_match(&text));
+        // todo: move file path to data module
         let text = "bitcoin/received_000000000001.csv.gz";
         assert!(r.is_match(&text));
+        // todo: move file path to data module
         let text = "bitcoin/address_000000000001.csv.gz";
         assert!(!r.is_match(&text));
     }
+
+    #[derive(Deserialize, std::fmt::Debug)]
+    pub struct Lotr {
+        src_id: String,
+        dst_id: String,
+        time: i64,
+    }
+
+    #[test]
+    fn test_headers_flag_and_delimiter() {
+        let g = GraphDB::new(2);
+        // todo: move file path to data module
+        let csv_path: PathBuf = [env!("CARGO_MANIFEST_DIR"), "resources/withheader"]
+            .iter()
+            .collect();
+
+        println!("path = {}", csv_path.as_path().to_str().unwrap());
+        let csv_loader = CsvLoader::new(Path::new(&csv_path));
+        let has_header = true;
+        let delimiter_string = ",";
+        let delimiter: u8 = delimiter_string.as_bytes()[0];
+
+        csv_loader
+            .set_header(has_header)
+            .set_delimiter(delimiter)
+            .load_into_graph(&g, |lotr: Lotr, g: &GraphDB| {
+                let src_id = calculate_hash(&lotr.src_id);
+                let dst_id = calculate_hash(&lotr.dst_id);
+                let time = lotr.time;
+
+                g.add_vertex(
+                    src_id,
+                    time,
+                    &vec![("name".to_string(), Prop::Str("Character".to_string()))],
+                );
+                g.add_vertex(
+                    dst_id,
+                    time,
+                    &vec![("name".to_string(), Prop::Str("Character".to_string()))],
+                );
+                g.add_edge(
+                    src_id,
+                    dst_id,
+                    time,
+                    &vec![(
+                        "name".to_string(),
+                        Prop::Str("Character Co-occurrence".to_string()),
+                    )],
+                );
+            })
+            .expect("Csv did not parse.");
+
+    }
+
+    #[test]
+    #[should_panic]
+    fn test_wrong_header_flag_file_with_header() {
+        let g = GraphDB::new(2);
+        // todo: move file path to data module
+        let csv_path: PathBuf = [
+            env!("CARGO_MANIFEST_DIR"),
+            "resources/withheader/",
+        ]
+        .iter()
+        .collect();
+        let csv_loader = CsvLoader::new(Path::new(&csv_path));
+        let has_header = false;
+        let delimiter_string = ",";
+        let delimiter: u8 = delimiter_string.as_bytes()[0];
+
+        csv_loader
```

Comment:
Since this piece of code is common across all the tests, we can move it to a function now!